### ISSUE TITLE
Layout::Document#export method return type is unspecified

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#564]
#### LayOut Ruby API Documentation Issue

[`Layout::Document#export`](https://ruby.sketchup.com/Layout/Document.html#export-instance_method) method does not document a return type, but the code snippets imply a status of some kind.

Does the method return immediately ?

 Ie, can we test for `File.exist?` on the exported pathname in the following statement, or would we need to wait a bit ?

----

ADD: Also, ... the YARD docs for the options hash are not properly done so as to list the options. (We have to examine the examples and surmise the argument types for the options.)

**Comment by thomthom** [comment]
I think the `status = ` is copy-pasta from older SketchUp examples.

The function doesn't have a useful return value. It returns `true` on success and raise a errors on failure.

Best if that example remove the use of `status =`.

**Comment by DanRathbun** [comment]
Yes I agree. In Testing I saw it return `true` and also experienced the `RangeError` when one of the pages was out of range. (Ie, a snippet had the pages as from 1 to 1, and there was only 1 page whose index was 0 in the document.)

**So ...** for example updates, I'd suggest:
* Showing the use of a `begin` ... `rescue` block wrapped around the use of this method. 
(I had to do this for my code because the errors cause subsequent features not to happen otherwise.)

* The docstring should also remind readers (with respect to the page options) that the pages array is zero based.

**Comment by DanRathbun** [comment]
I also see another potential problem. The [`Enumerable`](https://ruby-doc.org/core-2.5.5/Enumerable.html) module does **not** add in a `#last` method although it does add in a [`#first`](https://ruby-doc.org/core-2.5.5/Enumerable.html#method-i-first) method with expanded functionality.

So if I want to write a method for exporting all of the LayOut document's pages, how would it be best to supply the `end_page` index option ? Does the method allow an index of `-1` to signify the last page ?

The [`Layout::Pages`](https://ruby.sketchup.com/Layout/Pages.html) collection does have an [`#index`](https://ruby.sketchup.com/Layout/Pages.html#index-instance_method) method, but we cannot do ...
```ruby
doc.pages.index(pages.last)
```
... to get the last page's index.

So I suppose I'll have to use ...
```ruby
doc.pages.index(pages[-1])
```
... to be sure to pass a valid positive integer, and hope that the [`Layout::Pages#[]`](https://ruby.sketchup.com/Layout/Pages.html#[]-instance_method) method truly returns the last page in the collection with a `-1` index.

Ex:
```ruby
  # Export LayOut file to a PDF file.
  # The .pdf extensions will be automatically added to the filename.
  def export_layout(
    doc, # The Layout::Document object
    pg_start = 0,
    pg_end   = doc.pages.index(pages[-1]),
    export_dir = File.dirname(doc.path),
    filename = File.basename(doc.path, '.*')
  )
    #
    export_path = File.join(export_dir, "#{filename}.pdf")
    #
    doc.export(
      export_path,
      start_page: pg_start,
      end_page:   pg_end,
      output_resolution: Layout::PageInfo::RESOLUTION_HIGH,
      compress_images: true,
      compress_quality: 0.75
    )
  rescue => err
    puts "PDF export status: failed"
    puts err.inspect
  else
    puts "PDF export status: success"
    puts "Exported to: \"#{export_path}\""
  end
```

?
